predictsignauxfaibles/__main__.py

```diff
@@ -51,43 +86,37 @@ def evaluate(
     balanced_accuracy = balanced_accuracy_score(
         dataset.data["outcome"], model.predict(dataset.data)
     )
-    fbeta = fbeta_score(
-        dataset.data["outcome"], model.predict(dataset.data), beta=conf.EVAL_BETA
-    )
+    fbeta = fbeta_score(dataset.data["outcome"], model.predict(dataset.data), beta=beta)
     return {"balanced_accuracy": balanced_accuracy, "fbeta": fbeta}
 
 
-def run():  # pylint: disable=too-many-statements,too-many-locals
+def run(
+    args,
+):  # pylint: disable=too-many-statements,too-many-locals
     """
     Run model
     """
-    logging.info(
-        f"Running Model {conf.MODEL_ID} (commit {conf.MODEL_GIT_SHA}) ENV={conf.ENV}"
-    )
-    model_stats = {}
+    conf = load_conf(args)
+    logging.info(f"Running Model {conf.MODEL_ID} (commit {conf.MODEL_GIT_SHA})")
     model_id = datetime.now().strftime("%Y%m%d-%H%M%S")
+
+    datasets, model_stats = load_datasets_from_conf(args, conf)
+    (train_dataset, test_dataset, predict_dataset) = datasets
     model_stats["run_on"] = model_id
 
     step = "[TRAIN]"
     model_stats["train"] = {}
-    train_dataset = OversampledSFDataset(
-        conf.TRAIN_OVERSAMPLING,
-        date_min=conf.TRAIN_FROM,
-        date_max=conf.TRAIN_TO,
-        fields=conf.VARIABLES,
-        sample_size=conf.TRAIN_SAMPLE_SIZE,
-    )
-    logging.info(f"{step} - Fetching train set")
+    logging.info(f"{step} - Fetching train set ({train_dataset.sample_size} samples)")
     train_dataset.fetch_data()
 
     logging.info(f"{step} - Data preprocessing")
-    train_dataset.replace_missing_data().remove_na(ignore=IGNORE_NA)
+    train_dataset.replace_missing_data().remove_na(ignore=["time_til_outcome"])
```

Comment:
Let's keep `IGNORE_NA` from predictsignauxfaibles.config here so that we don't have to change this manually in __main__

Reply:
✅